open-metadata-implementation/access-services/asset-catalog/asset-catalog-server/src/main/java/org/odpi/openmetadata/accessservices/assetcatalog/handlers/AssetCatalogHandler.java

```diff
@@ -643,10 +643,7 @@ private List<EntityDetail> collectSearchedEntitiesByType(String userId,
             throws InvalidParameterException, PropertyServerException, UserNotAuthorizedException {
         List<EntityDetail> result = new ArrayList<>();
 
-        if(searchParameters.getExactMatch()) {
-            searchCriteria = repositoryHelper.getExactMatchRegex(searchCriteria, searchParameters.getCaseInsensitive());
-        }
-        else {
+        if(!searchParameters.getExactMatch()) {
             searchCriteria = repositoryHelper.getContainsRegex(searchCriteria, searchParameters.getCaseInsensitive());
```

Comment:
do we need to account for the case insensitive for the exact match case as well?

Reply:
- yes, hence the changes generic handler and not having the search string quoted twice (or regExed as I mentioned in the description). The value of the search string is modified according to ticked preferences
- only those that are not treated in the generic handler.